File: src/gtfs/models/ConvertConfig.py
from pydantic import BaseModel


class ConvertConfig(BaseModel):
    """Configuration for conversion process."""

    input_path: str
    output_path: str
    format: str = "binary"  # binary, json, both
    compression: bool = True
    debug_json: bool = False
    gen_transfers: bool = False
    allow_partial_trips: bool = False
    speed_walk: float = 1.33  # m/s
    transfer_cutoff: int = 500  # meters
    jobs: int = 1
    split_by_periods: bool = False  # Generate separate folders per service period
    gen_traces: bool = False  # Generate line geometry (lines.bin) from shapes.txt
    dry_run: bool = False  # Print the period plan without writing any files
    flat_output: bool = False  # Flat per-period files (routes_<period>.bin) vs subfolders
    write_index: bool = True  # Write index.bin/.json (disable with --no-index)
    pelo: bool = False  # Pelo preset: bare stops_/routes_<period>.bin at the root, nothing else

File: src/main/kotlin/com/raptor/gtfs/models/ConvertConfig.kt
package com.raptor.gtfs.models

import kotlinx.serialization.Serializable

@Serializable
data class ConvertConfig(
    val input_path: String,
    val output_path: String,
    val format: String = "binary",
    val compression: Boolean = true,
    val debug_json: Boolean = false,
    val gen_transfers: Boolean = false,
    val allow_partial_trips: Boolean = false,
    val speed_walk: Double = 1.33,
    val transfer_cutoff: Int = 500,
    val jobs: Int = 1,
    val split_by_periods: Boolean = false,
    val gen_traces: Boolean = false,
    val dry_run: Boolean = false,
    val flat_output: Boolean = false,
    val write_index: Boolean = true,
    val pelo: Boolean = false
)
